p2p/protocol/identify/obsaddr.go

```diff
@@ -232,10 +256,19 @@ func (oas *ObservedAddrManager) worker(ctx context.Context) {
 	defer ticker.Stop()
 
 	hostClosing := oas.host.Network().Process().Closing()
+	subChan := oas.reachabilitySub.Out()
 	for {
 		select {
+		case evt, ok := <-subChan:
+			if !ok {
+				subChan = nil
+			}
+			ev := evt.(event.EvtLocalReachabilityChanged)
+			oas.reachability = ev.Reachability
```

Comment:
this will panic when the channel gets closed, as evt will be nil.

Reply:
oops :/ Fixed.